pandas/sparse/array.py

```diff
@@ -769,14 +769,17 @@ def make_sparse(arr, kind='block', fill_value=None):
     if isnull(fill_value):
         mask = notnull(arr)
     else:
-        mask = arr != fill_value
+        mask = np.not_equal(arr, fill_value)
+        # In NumPy 1.12.0, not implemented at least for arr.dtype=str
+        if mask is NotImplemented:
+            mask = np.not_equal(arr.astype(object), fill_value)
```

Comment:
no we *don't* use ``not_equal`` at all it in fact doesn't work on object arrays (well maybe in 1.12 it does, but its not compat).

use ``pandas.types.missing.array_equivalent`` if you really need, though ``isnull`` isself already calls this.

again what are you trying to do compare an array w/o a scalar?